provider/cluster/manager.go

```diff
@@ -245,7 +248,35 @@ func (dm *deploymentManager) stopMonitor() {
 func (dm *deploymentManager) startDeploy() <-chan error {
 	dm.stopMonitor()
 	dm.state = dsDeployActive
-	return dm.do(dm.doDeploy)
+
+	chErr := make(chan error, 1)
+
+	go func() {
+		hostnames, err := dm.doDeploy()
+		if err != nil {
+			chErr <- err
+			return
+		}
+
+		if len(hostnames) != 0 {
+			// start update to takeover hostnames
+			dm.log.Info("hostnames withheld from deployment", "cnt", len(hostnames))
+		}
+
+		groupCopy := *dm.mgroup
+		ev := event.ClusterDeployment{
+			LeaseID: dm.lease,
+			Group:   &groupCopy,
+			Status:  event.ClusterDeploymentUpdated,
+		}
+		err = dm.bus.Publish(ev)
+		if err != nil {
+			dm.log.Error("failed publishing event", "err", err)
+		}
+
+		close(chErr)
```

Comment:
would prefer `defer close(chErr)` as the first statement to make it easier to read.

Reply:
I'll go back over this to see what I am doing with this code

Reply:
It's just a preferred pattern (there are more instances of this in this PR).

It makes reading the function a lot easier if you know that no matter what, this channel is going to be closed when it returns.

Also we aren't closing the channel when we send an error back here, which may be by-design inconsistency or not, but it adds more mental overhead when reading through it.